### ISSUE TITLE
Rotating logs to multiple files

### ISSUE THREAD

**Issue by madridu** [Delgan/loguru#244]
Hi!

I'm using loguru with an app that takes http petitions with flask and then does a task with it in a separate thread. When it does ok it logs with info and if there's an error, it logs it. I'm running it on an ubuntu.

On the time of rotation it opens a new file for every new thread and keeps on using it. For example, today I have 3 different files: Synclog.log, Synclog.2020-04-14_00-03-50_598472, Synclog.2020-04-14_00-08-56_524389 Since the rollover the app uses all of this 3 files to log.

```
logger.add('/home/web_hook/logs/Synclog.log', format='{time:DD/MM/YYYY HH:mm:ss} {level} {message}', level="INFO", rotation='00:00',retention='15 days',enqueue=True, encoding='utf8')

def do_the_task(http_petition):
      if task ok:
              logger.info(a message)
      else:
              logger.error(a message)

app = Flask(__name__)

@app.route("/")
def hello():
    if request.args:
        if request.args.get('action') == 'create':
            worker = Thread(target=do_the_task, args=(request.args,))
            worker.setDaemon(True)
            worker.start()
    return Response(status=200)
```
Is there a way to avoid this? any config I've missed? I've seen that you recommend using .bind in similar cases but not sure if it will work with my app.

Thanks!

**Comment by Delgan** [comment]
Hey @madridu, it seems `logger.add()` is called multiple times in your applications for some reasons. This is the reason why you are seeing three rotated files. At first, the three sinks logs to the same file so you won't notice anything. However, once rotation occurs, the sinks were not aware they shared the same file, so each one will open a new file independently. Your problem looks similar to #225.

You should make sure that `logger.add()` is not called more than once (if you want to log to the same file). This might be tricky if Flask or your application is using `multiprocessing` under the hood. Once the logger properly configured, rotation should work perfectly fine with any number of threads.

You can try to `print(logger)` from multiple places to see the currently added handlers. Do you see anything on your application that could cause `logger.add()` to be called thrice?

**Comment by madridu** [comment]
Hi @Delgan I saw #225 entry but it didn't look like my problem. I checked the code and couldn't find any trouble like that.

So looking for a multiprocessing problem which I couldn't find on my code, I introduced an `os.getpid() ` on every log entry to see if it was a problem with my server config and there were more than one instance of my app running. 

And it was indeed, gunicorn running flask app was launching 3 parallel processes. So I changed configuration to avoid this and voilà! All problems were gone.

Thanks for the help!
